grid-compose/internal/config/config.go

```diff
@@ -0,0 +1,130 @@
+package config
```

Comment:
this pkg could be named `parser` and could be in `pkg/` and should only be responsible for
- reading the config file
- one-to-one mapping to the config struct
- validate

another layer should do the mapping from the config struct to the deploymentData map with all the logic to create the missing fields 
- networks
- filter to get node id
- minimum resources if missing